File: app/src/main/java/sang/thai/tran/travelcompanion/fragment/DisplayUserInfoFragment.java
package sang.thai.tran.travelcompanion.fragment;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.hdodenhof.circleimageview.CircleImageView;
import sang.thai.tran.travelcompanion.R;
import sang.thai.tran.travelcompanion.activity.LoginActivity;
import sang.thai.tran.travelcompanion.activity.MainActivity;
import sang.thai.tran.travelcompanion.model.UserInfo;
import sang.thai.tran.travelcompanion.utils.ApplicationSingleton;
import sang.thai.tran.travelcompanion.view.EditTextViewLayout;

import static sang.thai.tran.travelcompanion.utils.AppConstant.NEED_SUPPORT_COMPANION;
import static sang.thai.tran.travelcompanion.utils.AppConstant.NEED_SUPPORT_COMPANION_GUIDE;
import static sang.thai.tran.travelcompanion.utils.AppConstant.NEED_SUPPORT_COMPANION_WELL;
import static sang.thai.tran.travelcompanion.utils.AppConstant.SUPPORT_COMPANION;
import static sang.thai.tran.travelcompanion.utils.AppConstant.SUPPORT_COMPANION_GUIDE;
import static sang.thai.tran.travelcompanion.utils.AppConstant.SUPPORT_COMPANION_WELL;
import static sang.thai.tran.travelcompanion.utils.DialogUtils.showAlertDialog;

public class DisplayUserInfoFragment extends BaseFragment {


    @BindView(R.id.tv_work_title)
    TextView tv_work_title;

    @BindView(R.id.ll_final_button)
    LinearLayout ll_final_button;

    @BindView(R.id.tv_update_info)
    TextView tv_update_info;

    @BindView(R.id.tv_register_flight_or_guide)
    TextView tv_register_flight_or_guide;

    @BindView(R.id.tv_register_well)
    TextView tv_register_well;

    @BindView(R.id.et_full_name)
    EditTextViewLayout et_full_name;

    @BindView(R.id.et_year_of_birth)
    EditTextViewLayout et_year_of_birth;

    @BindView(R.id.et_gender)
    EditTextViewLayout et_gender;

    @BindView(R.id.et_phone)
    EditTextViewLayout et_phone;

    @BindView(R.id.et_email)
    EditTextViewLayout et_email;

    @BindView(R.id.rlAdminAvatar)
    CircleImageView rlAdminAvatar;


    private String type;

    public static DisplayUserInfoFragment newInstance(Bundle bundle) {
        DisplayUserInfoFragment infoRegisterFragment = new DisplayUserInfoFragment();
        infoRegisterFragment.setArguments(bundle);
        return infoRegisterFragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_display_info, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
        initView();
    }

    private void initView() {
        Bundle intent = getArguments();
        if (intent != null) {
            String title = intent.getString(MainActivity.WORK_TITLE_EXTRA);
            type = intent.getString(MainActivity.USER_TYPE_EXTRA);
            if (!TextUtils.isEmpty(type)) {
                switch (type) {
                    case SUPPORT_COMPANION:
//                        title = getResources().getStringArray(R.array.list_item)[0];
                        title = getString(R.string.label_support);
                        tv_register_well.setVisibility(View.GONE);
                        break;
                    case SUPPORT_COMPANION_GUIDE:
//                        title = getResources().getStringArray(R.array.list_item)[1];
                        title = getString(R.string.label_support_guide);
                        tv_register_well.setVisibility(View.GONE);
                        tv_register_flight_or_guide.setText(getString(R.string.label_register_guide));
                        break;
                    case SUPPORT_COMPANION_WELL:
                        title = getString(R.string.label_support_well);
//                        title = getResources().getStringArray(R.array.list_item)[2];
                        tv_register_flight_or_guide.setText(getString(R.string.label_register_flight_companion_domestic));
                        tv_register_well.setText(getString(R.string.label_register_well_trained_companion));
                        break;
                    case NEED_SUPPORT_COMPANION:
                        title = getString(R.string.label_need_support);
                        tv_register_well.setVisibility(View.GONE);
                        ll_final_button.setVisibility(View.GONE);
                        break;
                    case NEED_SUPPORT_COMPANION_GUIDE:
                        title = getString(R.string.label_need_support_guide);
                        tv_register_well.setVisibility(View.GONE);
                        ll_final_button.setVisibility(View.GONE);
                        tv_register_flight_or_guide.setText(getString(R.string.label_register_guide));
                        break;
                    case NEED_SUPPORT_COMPANION_WELL:
                        title = getString(R.string.label_need_support_well);
                        ll_final_button.setVisibility(View.GONE);
                        tv_register_flight_or_guide.setText(getString(R.string.label_register_flight_companion));
                        tv_register_well.setText(getString(R.string.label_register_for_hour));
                        break;
                }
            }

            tv_update_info.setText(getString(R.string.label_update_info));
            tv_work_title.setText(title);
        }
        updateUserInfo();
    }

    private void updateUserInfo() {
        UserInfo userInfo = ApplicationSingleton.getInstance().getUserInfo();
        if (userInfo != null) {
            et_full_name.setText(userInfo.getFirst_Name());
            et_year_of_birth.setText(userInfo.getIdentify_Date());
            et_gender.setText(userInfo.getGender());
            et_phone.setText(userInfo.getPhone());
            et_email.setText(userInfo.getEmail());
            if (getActivity() != null)
            Glide.with(getActivity()).load(userInfo.getImage()).into(rlAdminAvatar);
        }
    }

    @OnClick(R.id.tv_update_info)
    protected void onClickUpdateInfo() {
        Intent intent = new Intent(getActivity(), LoginActivity.class);
        intent.putExtra(MainActivity.UPDATE_INFO, true);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }

    @OnClick(R.id.tv_register_flight_or_guide)
    protected void onClickRegisterFlight() {
        switch (type) {
            case SUPPORT_COMPANION:
                ((MainActivity) getActivity()).registerFlight(false);
                break;
            case SUPPORT_COMPANION_GUIDE:
                ((MainActivity) getActivity()).registerGuide(false);
                break;
            case SUPPORT_COMPANION_WELL:
                ((MainActivity) getActivity()).registerFlight(false);
                break;
            case NEED_SUPPORT_COMPANION:
                ((MainActivity) getActivity()).registerFlight(true);
                break;
            case NEED_SUPPORT_COMPANION_GUIDE:
                ((MainActivity) getActivity()).registerGuide(true);
                break;
            case NEED_SUPPORT_COMPANION_WELL:
                ((MainActivity) getActivity()).registerFlight(true);
                break;
        }
    }

    @OnClick(R.id.tv_register_well)
    protected void onClickRegisterGuide() {
        ((MainActivity) getActivity()).registerWell(type.equals(NEED_SUPPORT_COMPANION_WELL));
    }

    @OnClick(R.id.tv_done)
    protected void onDone() {
        showAlertDialog(getActivity(), getString(R.string.msg_finish_dialog), new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
    }

}

File: app/src/main/java/sang/thai/tran/travelcompanion/fragment/DisplayUserInfoFragment.kt
package sang.thai.tran.travelcompanion.fragment

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.fragment_display_info.*
import sang.thai.tran.travelcompanion.R
import sang.thai.tran.travelcompanion.activity.LoginActivity
import sang.thai.tran.travelcompanion.activity.MainActivity
import sang.thai.tran.travelcompanion.utils.AppConstant.*
import sang.thai.tran.travelcompanion.utils.ApplicationSingleton
import sang.thai.tran.travelcompanion.utils.DialogUtils.showAlertDialog
import sang.thai.tran.travelcompanion.utils.Log

class DisplayUserInfoFragment : BaseFragment() {

    private var type: String? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.d("Sang","onCreateView")
        return inflater.inflate(R.layout.fragment_display_info, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        Log.d("Sang","onActivityCreated: ${ApplicationSingleton.getInstance().userType}")
        tv_update_info.setOnClickListener { onClickUpdateInfo() }
        tv_register_flight_or_guide.setOnClickListener { onClickRegisterFlight() }
        tv_register_well.setOnClickListener { onClickRegisterGuide() }
        tv_done.setOnClickListener { onDone() }
        tv_list_need_support.setOnClickListener{showListOfNeedSupport()}
    }

    override fun onResume() {
        super.onResume()
        initView()
    }

    private fun initView() {

        var title = ""
        type = ApplicationSingleton.getInstance().userType
        Log.d("Sang", "initView: $type" )

        if (!TextUtils.isEmpty(type)) {
            when (type) {
                SUPPORT_COMPANION -> {
                    title = getString(R.string.label_support)
                    tv_register_well?.visibility = View.GONE
                }
                SUPPORT_COMPANION_GUIDE -> {
                    //                        title = getResources().getStringArray(R.array.list_item)[1];
                    title = getString(R.string.label_support_guide)
                    tv_register_well?.visibility = View.GONE
                    tv_register_flight_or_guide?.text = getString(R.string.label_register_guide)
                }
                SUPPORT_COMPANION_WELL -> {
                    title = getString(R.string.label_support_well)
                    //                        title = getResources().getStringArray(R.array.list_item)[2];
                    tv_register_flight_or_guide?.text = getString(R.string.label_register_flight_companion_domestic)
                    tv_register_well?.text = getString(R.string.label_register_well_trained_companion)
                    tv_register_flight_or_guide.visibility = View.GONE
                }
                NEED_SUPPORT_COMPANION -> {
                    title = getString(R.string.label_need_support)
                    tv_list_need_support?.visibility = View.GONE
                    tv_register_well?.visibility = View.GONE
                    ll_final_button?.visibility = View.GONE
                }
                NEED_SUPPORT_COMPANION_GUIDE -> {
                    title = getString(R.string.label_need_support_guide)
                    tv_list_need_support?.visibility = View.GONE
                    tv_register_well?.visibility = View.GONE
                    ll_final_button?.visibility = View.GONE
                    tv_register_flight_or_guide?.text = getString(R.string.label_register_guide)
                }
                NEED_SUPPORT_COMPANION_WELL -> {
                    title = getString(R.string.label_need_support_well)
                    tv_list_need_support?.visibility = View.GONE
                    ll_final_button?.visibility = View.GONE
                    tv_register_flight_or_guide?.text = getString(R.string.label_need_requester)
//                    tv_register_well?.text = getString(R.string.label_register_for_hour)
                    tv_register_well?.visibility = View.GONE
                }
            }
        }

        tv_update_info?.text = getString(R.string.label_update_info)
        tv_work_title?.text = title
        updateUserInfo()
    }

    private fun updateUserInfo() {
        val userInfo = ApplicationSingleton.getInstance().userInfo
        if (userInfo != null) {
            et_full_name?.text = userInfo.first_Name
            et_year_of_birth?.text = userInfo.birthday
            et_gender?.text = userInfo.gender
            et_phone?.text = userInfo.phone
            et_email?.text = userInfo.email
            if (activity != null)
                Glide.with(activity!!).load(userInfo.image).into(rlAdminAvatar!!)
        }
    }

    private fun onClickUpdateInfo() {
        val intent = Intent(activity, LoginActivity::class.java)
        intent.putExtra(MainActivity.UPDATE_INFO, true)
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
        startActivity(intent)
    }

    private fun onClickRegisterFlight() {
        when (type) {
            SUPPORT_COMPANION -> (activity as MainActivity).registerFlight(false)
            SUPPORT_COMPANION_GUIDE -> (activity as MainActivity).registerGuide(false)
            SUPPORT_COMPANION_WELL -> (activity as MainActivity).registerFlight(false)
            NEED_SUPPORT_COMPANION -> (activity as MainActivity).registerFlight(true)
            NEED_SUPPORT_COMPANION_GUIDE -> (activity as MainActivity).registerGuide(true)
            NEED_SUPPORT_COMPANION_WELL -> (activity as MainActivity).registerFlight(true)
        }
    }

    private fun onClickRegisterGuide() {
        (activity as MainActivity).registerWell(type == NEED_SUPPORT_COMPANION_WELL)
    }

    private fun onDone() {
        showAlertDialog(activity, getString(R.string.msg_finish_dialog)) { dialog, which -> dialog.dismiss() }
    }

    private fun showListOfNeedSupport() {
        (activity as MainActivity).showListNeedSupport(type)
    }

    companion object {
        fun newInstance(): DisplayUserInfoFragment {
            return DisplayUserInfoFragment()
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        Log.d("Sang","onSaveInstanceState")
    }
}
